tests/cupy_tests/indexing_tests/test_indexing.py

```diff
@@ -29,6 +31,15 @@ def test_take_no_axis(self, xp):
         b = xp.array([[10, 5], [3, 20]])
         return a.take(b)
 
+    # see cupy#3017
+    @testing.for_dtypes('bBhH')
```

Comment:
`for_int_dtypes`?
Char codes should in general be avoided because they depend on environment.

Reply:
Here I didn't use `for_int_dtypes` because I thought that it allocated too much memory for larger integer types, but, yes, now I think I can broadcast a tiny array to test this.

Reply:
I fixed the test to use `for_int_dtypes` with broadcasting arrays to keep off actual memory allocation. Tests with 64-bit integer types are skipped because of the error with too large dimensions. Would you take a look at it again?